### ISSUE TITLE
No symbols for file when methods are on a type defined in another file

### ISSUE THREAD

**Issue by ramya-rao-a** [sourcegraph/go-langserver#311]
In a package have 2 Go files

The first file:
```
package main

func main() {}
type foo int
```

The second file:
```
package main

func (*foo) Hello() {}
func (*foo) Bye()   {}
```
The symbol provider doesnt provide any symbols for the second file.

**Comment by ramya-rao-a** [comment]
> Do you mean that in both files the symbols request should return Hello and Bye?

No. What we saw was there were no symbols being returned in the second file.

@harikb Can you update your language server and try again? To avoid the case that you have an older version of the tool?

**Comment by harikb** [comment]
@ramya-rao-a , it does seem to work after I updated the language server. Although the UI appears slightly different. I guess that may be the difference with langage-server mode.

**With language-server**
![bc1](https://user-images.githubusercontent.com/952886/45110093-2f7d8a00-b0f6-11e8-8a0d-9a3886fd19d2.png)

**With-out language-server**
![bc2](https://user-images.githubusercontent.com/952886/45110099-31dfe400-b0f6-11e8-9adf-782ac26da5d9.png)

NOTE: The syntax error (red underline) shown on `package main` line is because of https://github.com/Microsoft/vscode-go/issues/1914 (unrelated to this issue). At the moment, vs code won't work for me until that is fixed. I will update this ticket after that spurious error is resolved.
